compiler/rustc_lexer/src/unescape.rs

```diff
@@ -145,15 +146,15 @@ impl Mode {
         }
     }
 
-    pub fn is_bytes(self) -> bool {
+    pub fn is_byte(self) -> bool {
         match self {
             Mode::Byte | Mode::ByteStr | Mode::RawByteStr => true,
             Mode::Char | Mode::Str | Mode::RawStr => false,
         }
     }
 }
 
-fn scan_escape(chars: &mut Chars<'_>, mode: Mode) -> Result<char, EscapeError> {
+fn scan_escape(chars: &mut Chars<'_>, is_byte: bool) -> Result<char, EscapeError> {
```

Comment:
This risks getting refactored to "let's introduce enum here for clarity" in the future :)

Looking at the match above, maybe we need to refine `Mode` to

```
enum Type { Char, Byte }
enum Style { Single(Type), Str(Type), RawStr(Type) }
```

(names TBD)

probably an overkill though.

Reply:
It's a fair observation. It would be annoying to have to write `unescape_literal(..., Mode::Str(Type::Char), ...)` though. I think I can live with the risk of later refactoring here :)